go.mod

```diff
@@ -5,6 +5,10 @@ go 1.24.0
 // Fix CVE-2022-28948.
 replace gopkg.in/yaml.v3 => gopkg.in/yaml.v3 v3.0.1
 
+// Pin semver to v3.3.0 to avoid breaking changes in v3.3.1
+// xref: https://github.com/fluxcd/source-controller/issues/1738
+replace github.com/Masterminds/semver/v3 => github.com/Masterminds/semver/v3 v3.3.0
+
```

Comment:
No need to pin it here, you can change it in require section. The pinning was needed for SC due to cosign depending on semver.